execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -394,6 +479,238 @@ where
+            // The parent is the node which pointed to the leaf we deleted, and that leaf had
+            // multiple siblings.
+            (None, Trie::Node { mut pointer_block }) if pointer_block.child_count() > 2 => {
+                let trie_node: Trie<K, V> = {
+                    pointer_block[idx as usize] = None;
+                    Trie::Node { pointer_block }
+                };
+                let trie_key = Blake2bHash::new(&trie_node.to_bytes()?);
+                new_elements.push((trie_key, trie_node))
+            }
+            // The parent is the node which pointed to the leaf we deleted, and that leaf had one or
+            // zero siblings.
+            (None, Trie::Node { mut pointer_block }) => {
+                let (sibling_idx, sibling_pointer) = match pointer_block
+                    .to_indexed_pointers()
+                    .find(|(jdx, _)| idx != *jdx)
+                {
+                    // There are zero siblings.  Elsewhere we maintain the invariant that only
+                    // the root node can contain a single leaf.  Therefore the parent is the
+                    // root node.  The resulting output is just the empty node and nothing else.
+                    None => {
+                        let trie_node = Trie::Node {
+                            pointer_block: Box::new(PointerBlock::new()),
+                        };
+                        let trie_key = Blake2bHash::new(&trie_node.to_bytes()?);
+                        new_elements.push((trie_key, trie_node));
+                        break;
+                    }
+                    Some((sibling_idx, pointer)) => (sibling_idx, pointer),
+                };
+                // There is one sibling.
+                match (sibling_pointer, parents.pop()) {
+                    (_, Some((_, Trie::Leaf { .. }))) => panic!("Should not have leaf in scan"),
+                    // There is no grandparent.  Therefore the parent is the root node. Output
+                    // the root node with the index zeroed out.
+                    // TODO: unify me
+                    (_, None) => {
+                        pointer_block[idx as usize] = None;
+                        let trie_node = Trie::Node { pointer_block };
+                        let trie_key = Blake2bHash::new(&trie_node.to_bytes()?);
+                        new_elements.push((trie_key, trie_node));
+                        break;
+                    }
+                    // The sibling is a leaf and the grandparent is a node.  Reseat the single leaf
+                    // sibling into the grandparent.
+                    (Pointer::LeafPointer(..), Some((idx, Trie::Node { mut pointer_block }))) => {
+                        pointer_block[idx as usize] = Some(sibling_pointer);
+                        let trie_node = Trie::Node { pointer_block };
+                        let trie_key = Blake2bHash::new(&trie_node.to_bytes()?);
```

Comment:
This is repeated a lot. Maybe we should add a `trie_node.compute_hash()` method?

Reply:
@henrytill and I agree we should do this; we will want to update all of the trie operations though.  Making a follow up ticket...